pkg/solana/logpoller/loader.go

```diff
@@ -249,7 +251,10 @@ func (c *EncodedLogCollector) loadSlotBlocksRange(ctx context.Context, start, en
 	}
 
 	for _, block := range result {
-		c.chBlock <- block
+		select {
+		case <-ctx.Done():
```

Comment:
Shouldn't there be a `break` statement here to break out of the for loop if the context is cancelled?

Reply:
Oh, also in this line at the beginning of the `loadSlotBlockRange` method where you add an rpc timeout... I think you should rename the child context to something else and use that only for the call to `GetBlocks`... passing the parent context here.

```
ctx, cancel := context.WithTimeout(ctx, c.rpcTimeLimit)
```

Reply:
That can provide assurance that the function exits immediately. I'll add it.

Reply:
I'm not worried about shadowing, I'm saying I don't agree with the functional behavior of the code--I'm suggesting that we use the child context only for the rpc call, and the parent context for the post processing.

The rpc timeout should not apply to the post processing after the rpc server has returned all the data.  There's no reason to abort the for loop unless the whole node is shutting down, since we know the for loop will be done soon--it doesn't depend on any network I/O.

Reply:
Bumping this in case you didn't see it earlier ^^ I was hoping you'd change this... or give an argument for why it's okay as is?

Reply:
As is, it's going to fetch data and then throw it away for no reason, which seems like the wrong behavior.

Reply:
Ahh I understand now. I agree and I'll make the change.